test/cache/cache-lock.test.ts

```diff
@@ -0,0 +1,367 @@
+import Redis from 'ioredis'
+import { ExecutionError } from 'redlock'
+import { Adapter, AdapterDependencies, AdapterEndpoint } from '../../src/adapter'
+import { RedisCache } from '../../src/cache'
+import { AdapterConfig } from '../../src/config'
+import { sleep } from '../../src/util'
+import { MockCache, NopTransport, RedisMock, TestAdapter } from '../../src/util/testing-utils'
+import { test } from './helper'
+
+test.serial(
+  'An adapter with a duplicate name and no cache prefix should fail to acquire a lock',
+  async (t) => {
+    const config = new AdapterConfig(
+      {},
+      {
+        envDefaultOverrides: {
+          CACHE_TYPE: 'redis',
+          CACHE_REDIS_PORT: 6000,
+          CACHE_LOCK_DURATION: 2000,
+          CACHE_LOCK_RETRIES: 2,
+        },
+      },
+    )
+    const adapter = new Adapter({
+      name: 'TEST',
+      defaultEndpoint: 'test',
+      config,
+      endpoints: [
+        new AdapterEndpoint({
+          name: 'nowork',
+          transport: new NopTransport(),
+        }),
+      ],
+    })
+
+    const config2 = new AdapterConfig(
+      {},
+      {
+        envDefaultOverrides: {
+          CACHE_TYPE: 'redis',
+          CACHE_REDIS_PORT: 6000,
+          CACHE_LOCK_DURATION: 2000,
+          CACHE_LOCK_RETRIES: 2,
+        },
+      },
+    )
+    const adapter2 = new Adapter({
+      name: 'TEST',
+      defaultEndpoint: 'test',
+      config: config2,
+      endpoints: [
+        new AdapterEndpoint({
+          name: 'nowork',
+          transport: new NopTransport(),
+        }),
+      ],
+    })
+
+    const redisClient = new RedisMock() as unknown as Redis
+    const cache = new RedisCache(redisClient) // Fake redis
+    const dependencies: Partial<AdapterDependencies> = {
+      cache,
+      redisClient,
+    }
+
+    try {
+      await TestAdapter.start(adapter, t.context, dependencies)
+      await TestAdapter.start(adapter2, t.context, dependencies)
+
+      t.fail('An ExecutionError should have been thrown')
+    } catch (error) {
+      if (error instanceof ExecutionError) {
+        t.is(error.message, 'The operation was unable to achieve a quorum during its retry window.')
```

Comment:
If this is the error the adapter throws, we should catch it and rethrow it with something easier to understand ("the adapter failed to acquire a lock on the cache, check the cache prefix and make sure that if you're running two instances... etc. etc."